sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/catalog/ExternalCatalog.scala

```diff
@@ -104,6 +104,8 @@ abstract class ExternalCatalog {
    */
   def alterTable(tableDefinition: CatalogTable): Unit
 
+  def alterTableSchema(db: String, table: String, schema: StructType): Unit
```

Comment:
please add some comments. One thing I wanna highlight is: the caller side is responsible for guarantee that the new schema still contains previous partition columns and bucket columns.